src/main/scala/edu/ie3/osmogrid/io/input/pbf/PbfWorker.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.osmogrid.io.input.pbf
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior, PostStop}
+import com.acervera.osm4scala.EntityIterator.fromBlob
+import com.acervera.osm4scala.model.RelationMemberEntityTypes.RelationMemberEntityTypes
+import com.acervera.osm4scala.model.{
+  Info,
+  NodeEntity,
+  RelationEntity,
+  RelationMemberEntity,
+  WayEntity
+}
+import edu.ie3.osmogrid.io.input.pbf.PbfGuardian.stopAndCleanup
+import edu.ie3.osmogrid.model.Osm4ScalaMapper.{osmNode, osmRelation, osmWay}
+import edu.ie3.osmogrid.model.OsmoGridModel.LvOsmoGridModel
+import edu.ie3.osmogrid.model.{OsmoGridModel, PbfFilter}
+import edu.ie3.util.geo.GeoUtils
+import edu.ie3.util.osm.model.OsmContainer
+import edu.ie3.util.osm.model.OsmContainer.ParOsmContainer
+import edu.ie3.util.osm.model.OsmEntity.{MetaInformation, Node, Relation, Way}
+import edu.ie3.util.osm.model.OsmEntity.Relation.{
+  RelationMember,
+  RelationMemberType
+}
+import org.openstreetmap.osmosis.osmbinary.fileformat.{Blob, BlobHeader}
+
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.collection.parallel.immutable.ParSeq
+
+private[pbf] object PbfWorker {
+
+  // external request protocol
+  sealed trait PbfWorkerMsg
+
+  final case class ReadBlobMsg(
+      blobHeader: BlobHeader,
+      blob: Blob,
+      filter: PbfFilter,
+      replyTo: ActorRef[PbfWorker.Response]
+  ) extends PbfWorkerMsg
+
+  // external response protocol
+  sealed trait Response
+
+  final case class ReadSuccessful(osmoGridModel: OsmoGridModel) extends Response
+
+  final case class ReadFailed(
+      blobHeader: BlobHeader,
+      blob: Blob,
+      filter: PbfFilter,
+      exception: Throwable
+  ) extends Response
```

Comment:
I might have missed some discussion here, but wouldn't it make sense to have these as part of `PbfGuardian` messages instead of wrapping them in `WrappedPbfWorkerResponse`?

Reply:
In this case the coupling between `PfGuardian` and `PbfWorker` is quite close, so there wouldn't be the necessity to couple the protocols more loosely. Still the concept brings the known flexibility. However, I don't see too much added value by making it a `PbfGuardianMessage`. Could you elaborate on that?

Reply:
Yes, we do gain some flexibility by using the wrapper here, but are we actually going to need this or is `PbfWorker` just always interacting with `PbfGuardian`? If so, we can save some quite a bit of code here